tests/UnitTests/CoreGraphics.drawing/CGContextDrawing_FlushTests.cpp

```diff
@@ -0,0 +1,132 @@
+    _CGContextPushBeginDraw(context);
+
+    CGContextSetRGBFillColor(context, 1, 0, 0, 1);
+    CGContextFillRect(context, bounds);
+
+    // Flush the red fill rect
+    CGContextFlush(context);
+
+    CGContextSetRGBFillColor(context, 0, 0, 1, 1);
+    CGContextFillRect(context, CGRectMake(0, 0, 300, 300));
+
+    // We should still have red & blue rectangle should not show up.
+    unsigned char* dataPtr = static_cast<unsigned char*>(CGBitmapContextGetData(context));
+    ASSERT_NE(dataPtr, nullptr);
+
+    // Validate only the red fill rect is executed.
+    EXPECT_EQ(dataPtr[0], 00);
+    EXPECT_EQ(dataPtr[1], 00);
+    EXPECT_EQ(dataPtr[2], 0xff);
+    EXPECT_EQ(dataPtr[3], 0xff);
+
+    CGContextFlush(context);
+
+    // We should now see the blue fill rect
+    EXPECT_EQ(dataPtr[0], 0xff);
+    EXPECT_EQ(dataPtr[1], 00);
+    EXPECT_EQ(dataPtr[2], 00);
+    EXPECT_EQ(dataPtr[3], 0xff);
+
+    // Add some extra drawings
+    CGContextClearRect(context, CGRectMake(100, 100, 200, 300));
+
+    CGPoint center = _CGRectGetCenter(bounds);
+
+    CGMutablePathRef concentricCirclesPath = CGPathCreateMutable();
+
+    CGPathAddEllipseInRect(concentricCirclesPath, nullptr, _CGRectCenteredOnPoint({ 50, 50 }, center));
+    CGPathAddEllipseInRect(concentricCirclesPath, nullptr, _CGRectCenteredOnPoint({ 100, 100 }, center));
+    CGPathAddEllipseInRect(concentricCirclesPath, nullptr, _CGRectCenteredOnPoint({ 150, 150 }, center));
+    CGPathAddEllipseInRect(concentricCirclesPath, nullptr, _CGRectCenteredOnPoint({ 200, 200 }, center));
+
+    CGContextSetRGBFillColor(context, 1.0, 0.0, 0.0, 0.5);
+    CGContextSetRGBStrokeColor(context, 1.0, 0.0, 0.0, 1.0);
+
+    CGContextAddPath(context, concentricCirclesPath);
+    CGContextDrawPath(context, kCGPathFillStroke);
+
+    CGPathRelease(concentricCirclesPath);
+
+    _CGContextPopEndDraw(context);
+}
+
+DRAW_TEST_F(CGContextFlush, FillFlushMultipleDrawingCounters, WhiteBackgroundTest<>) {
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+    static int sDrawCount = 5;
+
+    for (int i = 0; i < sDrawCount; ++i) {
+        _CGContextPushBeginDraw(context);
+    }
+
+    CGContextSetRGBFillColor(context, 1, 0, 0, 1);
+    CGContextFillRect(context, bounds);
+
+    for (int i = 0; i < sDrawCount; ++i) {
+        // Multiple flushes should work.
+        CGContextFlush(context);
+    }
+
+    // Add some extra drawings
+    CGContextClearRect(context, bounds);
+
+    // We should still have red & clear should not of been executed.
+    unsigned char* dataPtr = static_cast<unsigned char*>(CGBitmapContextGetData(context));
+    ASSERT_NE(dataPtr, nullptr);
+
+    // Validate only the red fill rect is executed.
+    EXPECT_EQ(dataPtr[0], 00);
+    EXPECT_EQ(dataPtr[1], 00);
+    EXPECT_EQ(dataPtr[2], 0xff);
+    EXPECT_EQ(dataPtr[3], 0xff);
+
+    for (int i = 0; i < 3; ++i) {
+        // Multiple flushes should work.
+        CGContextFlush(context);
```

Comment:
it's validated below, so no need here.

---
In reply to: [106073225](https://github.com/Microsoft/WinObjC/pull/2208#discussion_r106073225) [](ancestors = 106073225)